File: lib-mvpbase-appcompat/src/main/java/com/baozi/mvp/factory/FragmentFactory.java
package com.baozi.mvp.factory;

import android.support.v4.app.Fragment;

/**
 * Created by Administrator on 2017/8/8 0008.
 */

public class FragmentFactory {
    public static Fragment getFragment(Class<Fragment> zClass) {
        try {
            return zClass.newInstance();
        } catch (InstantiationException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: lib-mvpbase-kotlin/src/main/java/com/linfeng/mvp/factory/FragmentFactory.kt
package com.linfeng.mvp.factory

import android.support.v4.app.Fragment

/**
 * Created by Administrator on 2017/8/8 0008.
 */

object FragmentFactory {
    fun getFragment(zClass: Class<Fragment>): Fragment? {
        try {
            return zClass.newInstance()
        } catch (e: InstantiationException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }

        return null
    }
}
